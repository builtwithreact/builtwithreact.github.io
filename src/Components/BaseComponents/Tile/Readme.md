<!--- Sample with out markdown, one tab space is required -->
    <Tile />


<!--- Sample with out markdown -->

Sample Tile
```js
    <Tile />
```