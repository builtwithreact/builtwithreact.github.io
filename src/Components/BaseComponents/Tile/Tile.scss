.Tile {
  width: 32.5%;
  border: 1px solid rgba(0,0,0,.1)!important;
  border-radius: 2px;
  box-shadow: 0 1px 2px rgba(0,0,0,0.07);

  .Tile-Image {
    max-height: 190px;
    overflow: hidden;

    img {
      width: 100%;
      display: block;
    }
  }

  .Tile-Details {
    padding: 20px;

    .divider {
      margin: 0 70px 20px;
      border: none;
      border-bottom: 1px #e5e5e5 solid;
    }
  }

  .ProjectInfo {
    .ProjectInfo-Title {
      padding-bottom: 15px;
      color: rgba(0, 0, 0, .84);
      font-size: 21px;
      line-height: 24px;
      font-weight: 600;
      font-family: 'Barlow Semi Condensed', sans-serif;
    }

    .ProjectInfo-Description {
      color: rgba(0, 0, 0, .54);
      font-size: 14px;
      line-height: 18px;
    }
  }

  .AuthorInfo {
    font-size: 12px;
    display: flex;
    align-items: center;
    padding-top: 20px;
    a {
      color: rgba(0, 0, 0, .84);
      display: block;
      &:hover {
        text-decoration: underline;
      }
    }

    .AuthorInfo-Avatar {
      width: 40px;
      height: 40px;
      border-radius: 100%;
      border: 2px #4caf4f solid;
      margin-right: 20px;
    }

    .AuthorInfo-submissionDate {
      color: rgba(0, 0, 0, .54);
      padding-top: 5px;
      display: block;
    }
  }
  .Tile-CTA {
    background: #f8f8f8;
    border-top: 1px solid rgba(0,0,0,.1)!important;
    padding: 10px;
    justify-content: flex-end;
    display: flex;
    button {
      appearance: none;
      border: none;
      background: none;
      cursor: pointer;
      font-size: 14px;
      font-weight: normal;
      color: #aaa;
      &:hover {
       color: #777;
      }
      &:focus {
        outline: none;
      }
    }
  }
}