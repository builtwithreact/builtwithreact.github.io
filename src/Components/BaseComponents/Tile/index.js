import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Tile.css';
class Tile extends Component {
    render() {
        const { imageUrl, projectName, description, avatarUrl, authorName, submissionDate, gitHubUrl } = this.props
        return (
            <div className="Tile">
                <div className="Tile-Image">
                    <img src={imageUrl} alt={projectName} />
                </div>
                <div className="Tile-Details">
                    <hr className="divider" />
                    <div className="ProjectInfo">
                        <h3 className="ProjectInfo-Title"> {projectName} </h3>
                        <p className="ProjectInfo-Description">{description}</p>
                    </div>
                    <div className="AuthorInfo">
                        <a href={gitHubUrl}
                            target="_blank"
                            rel="noopener noreferrer">
                            <img className="AuthorInfo-Avatar"
                                src={avatarUrl}
                                alt={authorName} />
                        </a>
                        <div>
                            <a href={gitHubUrl}
                                target="_blank"
                                rel="noopener noreferrer">{authorName}</a>
                            <span className="AuthorInfo-submissionDate">{submissionDate}</span>
                        </div>
                    </div>
                </div>
                {/* <div className="Tile-CTA">
                    <button data-likes={this.props.likes} onClick={this.handleClickLikeBtn}>&hearts; {this.props.likes || 0}</button>
                    <button>&#128065; {this.props.views || 0}</button>
                </div> */}
            </div>
        );
    }
}

Tile.propTypes = {
    imageUrl: PropTypes.string,
    projectName: PropTypes.string,
    description: PropTypes.string,
    avatarUrl: PropTypes.string,
    authorName: PropTypes.string,
    gitHubUrl: PropTypes.string,
}

export default Tile;

