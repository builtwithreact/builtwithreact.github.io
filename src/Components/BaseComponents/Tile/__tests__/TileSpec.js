import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';
import Tile from './../index';

describe('Tile', ()=>{
    it('renders correctly', ()=>{
        let tree = renderer.create(
            <Tile  />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
