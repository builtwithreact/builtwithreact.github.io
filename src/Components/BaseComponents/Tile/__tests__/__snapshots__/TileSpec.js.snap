// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile renders correctly 1`] = `
<div
  className="Tile"
>
  <div
    className="Tile-Image"
    style={
      Object {
        "backgroundImage": "",
      }
    }
  >
    <img
      src="http://madewithreact.com/content/images/2017/03/react-logo-generator-1.png"
    />
  </div>
  <div
    className="Tile-Details"
  >
    <hr
      className="divider"
    />
    <div
      className="ProjectInfo"
    >
      <h3
        className="ProjectInfo-Title"
      >
         
         
      </h3>
      <p
        className="ProjectInfo-Description"
      />
    </div>
    <div
      className="AuthorInfo"
    >
      <a
        href="https://github.com/pankajladhar"
        target="_blank"
      >
        <img
          className="AuthorInfo-Avatar"
          name="Pankaj Ladhar"
          src="https://avatars2.githubusercontent.com/u/6375735?v=4"
        />
      </a>
      <div>
        <a
          href="https://github.com/pankajladhar"
          target="_blank"
        >
          Pankaj Ladhar
        </a>
        <span
          className="AuthorInfo-submissionDate"
        />
      </div>
    </div>
  </div>
</div>
`;
